src/types.ts

```diff
@@ -0,0 +1,72 @@
+export enum ComponentType {
+  Flow = 1,
+  Result = 3,
+  TaskList = 7,
+  Notice = 8,
+  FindProperty = 9,
+  DrawBoundary = 10,
+  PlanningConstraints = 11,
+  PropertyInformation = 12,
+  Question = 100, // Statement/DropDown
+  Checklist = 105,
+  TextInput = 110,
+  DateInput = 120,
+  AddressInput = 130,
+  ContactInput = 135,
+  FileUpload = 140,
+  MultipleFileUpload = 145,
+  NumberInput = 150,
+  Answer = 200, // Response
+  Content = 250,
+  InternalPortal = 300,
+  ExternalPortal = 310,
+  Section = 360,
+  SetValue = 380,
+  Pay = 400,
+  Filter = 500,
+  Review = 600,
+  Send = 650,
+  Calculate = 700,
+  Confirmation = 725,
+}
+
+export type Edges = Array<string>;
+
+export interface Node {
+  id?: string;
+  type?: number;
+  edges?: Edges;
+  data?: Record<string, any>;
+}
+
+export type Flow = {
+  _root: {
+    edges: Edges;
+  };
+  [key: string]: Node;
+};
+
+export interface IndexedNode extends Node {
+  id: string;
+}
+
+export type OrderedFlow = Array<IndexedNode>;
+
+export interface Crumb {
```

Comment:
`Crumb` is actually a rename of the `UserData` type which I think fits better semantically within a collection of `Breadcrumbs` (plural because it's a collection - a dictionary in the case of `Breadcrumbs` and an array in the case of `OrderedBreadcrumbs`).

Maybe we can agree a consistent naming convention as part of the data schema design exercise? I'd like to introduce a `ResponseData` and `Responses` type in place of both of these but that's a job/discussion for another day.